.github/workflows/automatic-cherry-pick.yaml

```diff
@@ -0,0 +1,246 @@
+          fi
+        
+      - name: Check Previous Cherry-picks
+        if: env.ERROR_MSG == ''
+        run: |
+          # Check that branches with the cherry-picked commits have not been pushed to the remote yet
+          duplicated_branches=""
+          for branch in ${{ steps.command.outputs.branches }}; do
+            new_branch=cherry_pick_from_pr${{ github.event.issue.number }}_into_$branch
+            if [[ -z "$(git ls-remote --heads origin $new_branch)" ]]; then
+              echo "No previous attempt to cherry-pick commits from this PR into branch $branch found."
+            else
+              duplicated_branches="$duplicated_branches $branch"
+            fi
+          done
+          if [[ -n "$duplicated_branches" ]]; then
+            errors="It seems there are previous unfinished attempts to cherry-pick commits from this PR to the following branch(es):"
+            for branch in $duplicated_branches; do
+              errors="$errors
+           - [$branch](https://${{ github.repository }}/tree/$branch)"
+            done
+            errors="$errors
+          
+          If the current cherry-pick attempt is for a different set of commits, make sure that the previous attempts are fully merged and that the corresponding branches have been deleted."
+             printf "ERROR_MSG<<EOF\n%s\nEOF" "$errors" >> $GITHUB_ENV
+          fi
+
+      - name: Status Report
+        id: report
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          if [[ -n '${{ env.ERROR_MSG }}' ]]; then
+            body="> ${{ steps.command.outputs.command }}
+          
+          Automatic cherry-pick failed. ${{ env.ERROR_MSG }}"
+            gh pr comment ${{ github.event.issue.number }} --body "$body"
+            echo "status=failure" >> $GITHUB_OUTPUT
+          else
+            echo "status=success" >> $GITHUB_OUTPUT
+          fi
+
+  create_pr:
+    runs-on: ubuntu-latest 
+    needs: process_comment
+    if: needs.process_comment.outputs.status == 'success'
+    permissions:
+      contents: write
+      pull-requests: write
+    strategy:
+      matrix:
+        branch: ${{ fromJson(needs.process_comment.outputs.branch_matrix) }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # We need the history of all branches
+          fetch-depth: 0
+
+      - name: Determine branch information
+        id: info
+        run: |
+          echo "new_branch=cherry_pick_from_pr${{ github.event.issue.number }}_into_${{ matrix.branch }}" >> $GITHUB_OUTPUT
+          echo "target_branch_url=[${{ matrix.branch }}](https://github.com/${{ github.repository }}/tree/${{ matrix.branch }})" >> $GITHUB_OUTPUT
+ 
+      - name: Cherry-pick commits
+        id: cherry-pick
+        continue-on-error: true
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
Note that if your repository requires signed commits to merge pull requests, commits authored by `github-actions[bot]` are unverified. At ACCESS-NRI I'm currently trying to find a way to put the required credentials for `access-bot[bot]` to have verified commits.

Reply:
That's good to know. In this case the commits will be generated by a cherry-pick command, which means that each commit has an author (the person that created the original commit) and a committer (the bot). I wonder who is required to sign the commit. Both?

Reply:
Answering myself: the bot also needs to sign, even if the original commit was signed.

Reply:
Added a comment about this so people are aware of this limitation.